omx/gstomxscaler.c

```diff
@@ -0,0 +1,376 @@
+ *
+ */
+
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <gst/gst.h>
+
+#include "OMX_TI_Common.h"
+#include <omx_vfpc.h>
+#include <OMX_TI_Index.h>
+
+#include "gstomxscaler.h"
+
+GST_DEBUG_CATEGORY_STATIC (gst_omx_scaler_debug_category);
+#define GST_CAT_DEFAULT gst_omx_scaler_debug_category
+
+/* prototypes */
+static void gst_omx_scaler_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec);
+static void gst_omx_scaler_get_property (GObject * object, guint prop_id,
+    GValue * value, GParamSpec * pspec);
+static GstCaps *gst_omx_scaler_transform_caps (GstOMXVideoFilter * self,
+    GstPadDirection direction, GstPad * srcpad, GstCaps * caps,
+    GstCaps * filter);
+static gboolean gst_omx_scaler_set_format (GstOMXVideoFilter * videofilter,
+    GstCaps * incaps, GstVideoInfo * ininfo, GList * outcaps_list,
+    GList * outinfo_list);
+static GstCaps *gst_omx_scaler_fixed_src_caps (GstOMXVideoFilter * self,
+    GstCaps * incaps, GstPad * srcpad);
+static GstCaps *gst_omx_scaler_fixate_caps (GstOMXVideoFilter * self,
+    GstPad * srcpad, GstCaps * sinkcaps, GstCaps * srccaps);
+enum
+{
+  PROP_0,
+};
+
+#define GST_OMX_SCALER_MAX_WIDTH    1920
+#define GST_OMX_SCALER_MAX_HEIGHT   1080
+#define GST_OMX_SCALER_MIN_WIDTH    16
+#define GST_OMX_SCALER_MIN_HEIGHT   16
+
+/* class initialization */
+
+#define DEBUG_INIT \
+  GST_DEBUG_CATEGORY_INIT (gst_omx_scaler_debug_category, "omxscaler", 0, \
+      "debug category for gst-omx video scaler");
+
+G_DEFINE_TYPE_WITH_CODE (GstOMXScaler, gst_omx_scaler,
+    GST_TYPE_OMX_VIDEO_FILTER, DEBUG_INIT);
+
+static void
+gst_omx_scaler_class_init (GstOMXScalerClass * klass)
+{
+  GObjectClass *gobject_class = G_OBJECT_CLASS (klass);
+  GstElementClass *element_class = GST_ELEMENT_CLASS (klass);
+  GstOMXVideoFilterClass *videofilter_class =
+      GST_OMX_VIDEO_FILTER_CLASS (klass);
+
+  gobject_class->set_property = gst_omx_scaler_set_property;
+  gobject_class->get_property = gst_omx_scaler_get_property;
+
+  videofilter_class->num_outputs = 1;
+  videofilter_class->cdata.default_src_template_caps = "video/x-raw,"
+      "format=(string)YUY2, width=(int) [ 16, 1920 ], "
+      "height=(int) [ 16, 1080 ], " "framerate = " GST_VIDEO_FPS_RANGE;
+  videofilter_class->cdata.default_sink_template_caps =
+      "video/x-raw, " "format=(string)NV12,  width=(int) [ 16, 1920 ], "
+      "height = (int) [ 16, 1200 ], framerate = " GST_VIDEO_FPS_RANGE;
+  videofilter_class->fixed_src_caps =
+      GST_DEBUG_FUNCPTR (gst_omx_scaler_fixed_src_caps);
+  videofilter_class->set_format = GST_DEBUG_FUNCPTR (gst_omx_scaler_set_format);
+  gst_element_class_set_static_metadata (element_class,
+      "OpenMAX Video Scaler",
+      "Filter/Encoder/Video",
+      "Scale raw video streams",
+      "Melissa Montero <melissa.montero@ridgerun.com>");
+}
+
+static void
+gst_omx_scaler_init (GstOMXScaler * self)
+{
+
```

Comment:
Shoudn't calls like this have the explicit return call?

Reply:
I think it's part of the GObject standard.

Reply:
Yes it is, just wondering if the function should be defined explicitely when is empty.